Automattic/wp-calypso
# Backup: Redirect from backup plugin to wp-admin on connect when site has a backup plan

This PR aims to fix an unexpected scenario that redirects a secondary admin to the `Explore our Jetpack Plans` pages after log in, even when the site already has a backup plan detected on https://github.com/Automattic/jetpack/pull/30862.

## Proposed Changes

* Add a condition to redirect back to wp-admin when the authentication request comes from the backup plugin and the site already [has a backup plan](https://github.com/Automattic/wp-calypso/blob/31bee99a3e842f1dbb85edee8771476eff3e95f6/client/jetpack-connect/authorize.js#LL1001C4-L1001C87).

| Before | After |
|---|---|
| <img width="500" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/2fb36b7b-60e3-4db4-9f98-ff2fd80682be"> | <img width="500" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/cb9c26a8-5187-43c5-b5e0-b4950ae14f37"> |

## Demo

https://github.com/Automattic/wp-calypso/assets/1488641/9d03eed0-514c-4056-a6f5-d6c6e32c892f

## Testing Instructions
* Checkout this branch locally and run Calypso by using`yarn start`.
* Spin up a Jurassic Ninja site and install the [Jetpack VaultPress Backup plugin 1.8-beta](https://github.com/Automattic/jetpack-backup-plugin/releases/tag/1.8-beta).
* Connect the site to Jetpack using your main WordPress.com account and add a Jetpack VaultPress Backup plan.
* Add a new user to your blog.
* In a separate browser (you could use a new profile in Google Chrome), log in with the new user.
* Navigate to VaultPress Backup page
* You should see a connect screen without any pricing info (this is because the site already has a backup plan). Click on `Log in`. 
* You should be redirected to Jetpack auth page.
* Log in using a different WordPress.com account than the main user. Do not click on `Approve` yet.
* Change the URL from `https://wordpress.com` to `http://calypso.localhost:3000` and refresh the page.
* You should continue see your user authenticated with the `Approve` button.
* Click on `Approve`.
* After the authorization completes, ensure you are redirected to your site backup page (wp-admin).

## Pre-merge Checklist
